internal/k8s/service/resolver.go

```diff
@@ -182,31 +182,28 @@ func (r *ResolvedReference) Item() client.Object {
 }
 
 type BackendResolver interface {
-	Resolve(ctx context.Context, ref gwv1alpha2.BackendObjectReference) (*ResolvedReference, error)
+	Resolve(ctx context.Context, namespace string, ref gwv1alpha2.BackendObjectReference) (*ResolvedReference, error)
 }
 
 type backendResolver struct {
-	namespace string
-	client    gatewayclient.Client
-	consul    *api.Client
-	logger    hclog.Logger
-	mapper    common.ConsulNamespaceMapper
+	client gatewayclient.Client
+	consul *api.Client
+	logger hclog.Logger
+	mapper common.ConsulNamespaceMapper
 }
 
-func NewBackendResolver(logger hclog.Logger, namespace string, mapper common.ConsulNamespaceMapper, client gatewayclient.Client, consul *api.Client) BackendResolver {
+func NewBackendResolver(logger hclog.Logger, mapper common.ConsulNamespaceMapper, client gatewayclient.Client, consul *api.Client) *backendResolver {
```

Comment:
Removing the `namespace` from the resolver itself and instead providing it as an arg to `Resolve()` lets us re-use one resolver for backends across different namespaces 💯